tests/helpers/include/data_format_inference.h

```diff
@@ -0,0 +1,51 @@
+#include "tensix_types.h"
+
+#ifdef ARCH_WORMHOLE
+const bool ARCH_BLACKHOLE = false;
+#endif
+
+#ifdef ARCH_BLACKHOLE
+const bool ARCH_WORMHOLE = true;
+#endif
+
+struct Formats {
+    const uint32_t unpack_src; 
+    const uint32_t unpack_dst;
+    const uint32_t pack_src;
+    const uint32_t pack_dst;
+};
+
+constexpr bool is_exponentB(uint32_t format){
+    return (format == static_cast<uint32_t>(DataFormat::Float16_b) || format == static_cast<uint32_t>(DataFormat::Bfp8_b) || format == static_cast<uint32_t>(DataFormat::Tf32));
+}
+
+constexpr bool format_combo_is_outlier(uint32_t input, uint32_t output, bool is_fp32_dest_acc_en){
```

Comment:
```suggestion
constexpr bool format_combo_is_outlier(DataFormat input, DataFormat output, bool is_fp32_dest_acc_en){
```
Let's try to be more expressive with the `DataFormat` enum. That way, this function becomes:

```cpp
constexpr bool format_combo_is_outlier(DataFormat input, DataFormat output, bool is_fp32_dest_acc_en) {
    return is_exponentB(input) && output == DataFormat::Float16 && !is_fp32_dest_acc_en;
}
```

Reply:
Also, can we find a better name for this method, like `is_format_outlier` or something similar?

Reply:
For now it's an outlier because the current combination doesn't work and there is no cohesive logic/explanation as to why. But we expect it to be resolved with HW team and then we will have no need for this function, at which point it will be removed. I can change the name of the function or add a comment explaining the situation.

Reply:
That would do.